app/src/main/java/br/org/cesar/discordtime/stickysessions/ui/lobby/SessionTypeAdapter.java

```diff
@@ -0,0 +1,130 @@
+package br.org.cesar.discordtime.stickysessions.ui.lobby;
+
+import android.content.Context;
+import android.graphics.Color;
+import android.graphics.ColorFilter;
+import android.graphics.PorterDuff;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import java.util.List;
+
+import androidx.recyclerview.widget.RecyclerView;
+import br.org.cesar.discordtime.stickysessions.R;
+import br.org.cesar.discordtime.stickysessions.domain.model.SessionType;
+import io.reactivex.Observable;
+import io.reactivex.subjects.PublishSubject;
+
+public class SessionTypeAdapter extends RecyclerView.Adapter<SessionTypeAdapter.ViewHolder> {
+
+    private Context mContext;
+    private List<SessionTypeViewModel> mSessionTypes;
+
+    private PublishSubject<SessionType> clickSubject;
+    Observable<SessionType> clickEvent;
+
+    SessionTypeAdapter(Context context, List<SessionTypeViewModel> sessionTypes) {
+        mContext = context;
+        mSessionTypes = sessionTypes;
+        clickSubject = PublishSubject.create();
+        clickEvent = clickSubject;
+    }
+
+    @Override
+    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View itemView = LayoutInflater
+                .from(parent.getContext())
+                .inflate(R.layout.lobby_card_view, parent, false);
+        return new ViewHolder(itemView);
+    }
+
+    @Override
+    public void onBindViewHolder(ViewHolder holder, int position) {
```

Comment:
Shouldn't the adapter be a passive view as described in https://android.jlelse.eu/recyclerview-in-mvp-passive-views-approach-8dd74633158?

Reply:
We can do it by creating another presenter, but list here is fixed and not retrieved from any database. Let's discuss that.